databricks/koalas/internal.py

```diff
@@ -416,6 +416,13 @@ def __init__(self, sdf: spark.DataFrame,
         else:
             self._column_index_names = column_index_names
 
+    def __delitem__(self, key):
+        self._sdf = self._sdf.drop(key)
+        self._data_columns.remove(key)
+        for i, col in enumerate(self._column_index):
+            if col[0] == key:
+                self._column_index.pop(i)
```

Comment:
Please keep `_InternalFrame` as immutable. It has some cache internally, so if we break the immutability, then the cache doesn't make sense anymore.

Reply:
@ueshin , Thanks for review! okay i understand it.
Anyway in pandas, they use a method named `_reset_cacher` (maybe reset the cache literally) when they delete column from DataFrame with `pop` method like below.

<img width="211" alt="스크린샷 2019-09-18 오전 8 55 08" src="https://user-images.githubusercontent.com/44108233/65087805-0a527580-d9f2-11e9-926b-cc24dc573f4a.png">

maybe are we have a similar thing for reset the internal cache? or for manually set & unset cache?

Thanks for review again, Have a good day :)

Reply:
What is a proper way to do this (remove a specific key and corresponding data)? Replace the current `_InternalFrame` with a new one like below?

```
internal = DataFrame._internal.copy(new_sdf, **new_metadata)
DataFrame._internal = internal
```

Reply:
Yes. Currently the proper way is:

```python
internal = DataFrame._internal.copy(new_sdf, **new_metadata)
DataFrame(internal)
```

or

```python
internal = _InternalFrame(sdf=...)
DataFrame(internal)
```

Reply:
@itholic, let's stick to the way we're doing for now. Cache stuff is orthogonal with the current API.

Reply:
@harupy @HyukjinKwon 
Thank you guys! they're really helpful for me. i'm going to fix them right away.